source/numerics.tex

```diff
@@ -3971,28 +3971,47 @@
 
 \indexlibraryglobal{generate_canonical}%
 \begin{itemdecl}
-template<class RealType, size_t bits, class URBG>
+template<class RealType, size_t digits, class URBG>
   RealType generate_canonical(URBG& g);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects
- Invokes \tcode{g()} $k$ times
- to obtain values $g_0, \dotsc, g_{k-1}$, respectively.
- Calculates a quantity
+Let
+\begin{itemize}
+\item $r$ be \tcode{numeric_limits<RealType>::radix},
+\item $R$ be $\tcode{g.max()} - \tcode{g.min()} + 1$,
+\item $d$ be the smaller of
+  \tcode{digits} and \tcode{numeric_limits<RealType>::digits},
+  \begin{footnote}
+  $d$ is introduced to avoid any attempt
+  to produce more bits of randomness
+  than can be held in \tcode{RealType}.
+  \end{footnote}
+\item $k$ be the smallest integer such that $R^k \ge r^d$, and
+\item $x$ be $\left\lfloor R^k / r^d \right\rfloor$.
+\end{itemize}
+An \defn{attempt} is $k$ invocations of \tcode{g()}
```

Comment:
How about leaving the "An _attempt_ is _k_ invocations ..." paragraph as part of the effects, and combining the _Returns_: into the _Effects_:, like so:

> _Effects_: An _attempt_ is _k_ invocations of `g()` to obtain values [...]. Attempts are made until _S < xr<sup>d</sup>_, then returns [...].